### PR TITLE
Update the version to 3.4.0-dev and sync changes from 3.3 branch

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

**Review by hlandau — APPROVED** [review]

File: NEWS.md [hunk 1]
```diff
@@ -18,14 +18,135 @@ OpenSSL Releases
...
+
+OpenSSL 3.4.0 is a feature release adding significant new functionality to
+OpenSSL.
+
+This release is in development.
+
 OpenSSL 3.3
 -----------
 
 ### Major changes between OpenSSL 3.2 and OpenSSL 3.3 [under development]
```
Comment (slontis) [inline NEWS.md:34]: Should [under development]  be here?

**Review by slontis — COMMENTED** [review]
Are there supposed to be multiple entries in CHANGES.md containing the date string
as `[xx XXX xxxx]`

File: NEWS.md (hunk 1 above)
Comment (t8m) [reply NEWS.md:34]: Well, the date will be Apr 17th but it was not released yet. We can sync it later.

**Comment by t8m** [comment]
> Are there supposed to be multiple entries in CHANGES.md containing the date string as `[xx XXX xxxx]`

I do not think it matters much. We fix it up later once 3.3.0 is released.

**Review by nhorman — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for the reviews.

**Pull request closed without merging**